tcms/kiwi_auth/views.py

```diff
@@ -27,65 +29,53 @@ class PasswordResetView(views.PasswordResetView):
-                )
-                messages.add_message(
-                    request,
-                    messages.INFO,
-                    _('Following is the administrator list')
-                )
-
-                # super-users can approve others
-                for user in User.objects.filter(is_superuser=True):
-                    messages.add_message(
-                        request,
-                        messages.INFO,
-                        '<a href="mailto:{}">{}</a>'.format(user.email,
-                                                            user.get_full_name() or user.username)
-                    )
-
-                # site admins should be able to do so too
-                for name, email in settings.ADMINS:
-                    messages.add_message(
-                        request,
-                        messages.WARNING,
-                        '<a href="mailto:{}">{}</a>'.format(email, name)
-                    )
-
-            return HttpResponseRedirect(reverse('core-views-index'))
-    else:
-        form = forms.RegistrationForm()
-
-    context_data = {
-        'form': form,
-    }
-    return render(request, 'registration/registration_form.html', context_data)
+    template_name = 'registration/registration_form.html'
+    form_class = RegistrationForm
+    success_url = reverse_lazy('core-views-index')
+
+    def post(self, request, *args, **kwargs):
+        form = self.form_class(data=request.POST, files=request.FILES)
+        if not form.is_valid():
+            return None
+
+        new_user = form.save()
+        activation_key = form.set_activation_key()
+        # send a signal that new user has been registered
+        USER_REGISTERED_SIGNAL.send(sender=form.__class__, request=request, user=new_user)
+
+        # Send confirmation email to new user
+        if settings.DEFAULT_FROM_EMAIL and settings.AUTO_APPROVE_NEW_USERS:
+            form.send_confirm_mail(request, activation_key)
+
+            message = 'Your account has been created, please check your mailbox for confirmation'
+            messages.add_message(request, messages.SUCCESS, _(message))
+        else:
+            message = 'Your account has been created, but you need an administrator to activate it'
+            messages.add_message(request, messages.WARNING, _(message))
+
+            messages.add_message(request, messages.INFO, _('Following is the administrator list'))
+            self.show_super_users_emails(request)
+            self.show_site_admins_emails(request)
+
+        return HttpResponseRedirect(self.success_url)
+
+    def show_site_admins_emails(self, request):
+        # site admins should be able to do so too
+        for name, email in settings.ADMINS:
+            mailto = '<a href="mailto:{}">{}</a>'.format(email, name)
+            messages.add_message(request, messages.WARNING, mailto)
+
+    def show_super_users_emails(self, request):
```

Comment:
while you are at it, these methods do not use `self` and pylint complains about them. You can make them `@staticmethod` to make pylint happy.

Reply:
Sure. Done.